app/src/org/commcare/activities/GlobalPrivilegeClaimingActivity.java

```diff
@@ -101,72 +105,45 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         switch(requestCode) {
             case BARCODE_CAPTURE:
                 if (resultCode == RESULT_OK) {
-                    String[] fields = processScanResult(data.getStringExtra("SCAN_RESULT"));
-                    if (fields == null) {
-                        privilegeClaimFailed();
-                    } else {
-                        String flag = fields[0];
-                        String username = fields[1];
-                        String signature = fields[2];
-                        if (checkProperFormAndAuthenticity(flag, username, signature)) {
-                            GlobalPrivilegesManager.enablePrivilege(flag, username);
-                            refreshUI();
-                        } else {
-                            privilegeClaimFailed();
+                    String scanResult = data.getStringExtra("SCAN_RESULT");
+                    try {
+                        Pair<String, String[]> activatedPrivileges =
+                                new PrivilegesUtility(GlobalConstants.TRUSTED_SOURCE_PUBLIC_KEY).
+                                        processPrivilegePayloadForActivatedPrivileges(scanResult);
+
+                        for(String p : activatedPrivileges.second) {
+                            if (!GlobalPrivilegesManager.allGlobalPrivilegesList.contains(p)) {
```

Comment:
Honestly no strong feelings about it specifically, other than that coding in behavior we don't actually use could have unintended consequences. Happy to pull this out and just let the phone process ones it doesn't understand.

The only argument for a whitelist would be that in theory it kind of makes sense for the phone to wipe all *not* present privileges, which it can't do if it doesn't know the keys for all of the ones that have been activated. 

Anyway, happy to go either direction